### PR TITLE
Tecplot ascii data reader

### PR DESCRIPTION
**Pull request by mechajeff**
### Overview
This feature adds a Tecplot ascii .dat file reader.  Tecplot is a common format for computational fluid dynamics and finite element model data.

### Details
Adds Tecplot ascii format data files to the list of PyVista readers.  The reader creates a MultiBlock data object of the information.  A tecplot example file is included in vtk_data.  A test was inclued in tests/utilites/test_reader.py

### CODE REVIEW

**Review by mechajeff — COMMENTED** [review]
Edit to reader.py TecplotReader class docstring.

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -3940,6 +3940,41 @@ def download_cgns_structured(load=True):  # pragma: no cover
     return pyvista.get_reader(filename).read()
 
 
+def download_tecplot_ascii(load=True):  # pragma: no cover
+    """Download the single block ascii Tecplot dataset.
```
Comment (tkoyama010) [inline pyvista/examples/downloads.py:3944]:
```suggestion
    """Download the single block ASCII Tecplot dataset.
```

File: pyvista/utilities/reader.py [hunk 2]
```diff
@@ -1142,6 +1144,23 @@ class STLReader(BaseReader):
...
+    """Tecplot Reader for ascii .dat files.
+
+    Examples
+    --------
+    # >>> import pyvista
+    # >>> from pyvista import examples
+    # >>> filename = examples.download_tecplot_ascii(load=False)
+    # >>> reader = pyvista.get_reader(filename)
+    # >>> mesh = reader.read()
+    # >>> mesh[0].plot()
```
Comment (tkoyama010) [inline pyvista/utilities/reader.py:1157]:
```suggestion
    >>> import pyvista
    >>> from pyvista import examples
    >>> filename = examples.download_tecplot_ascii(load=False)
    >>> reader = pyvista.get_reader(filename)
    >>> mesh = reader.read()
    >>> mesh[0].plot()
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you for your contribution! LGTM with small suggestions.

**Review by mechajeff — COMMENTED** [review]
test updated and ASCII type corrected

**Review by tkoyama010 — DISMISSED** [review]

File: pyvista/examples/downloads.py [hunk 3]
```diff
@@ -3940,6 +3940,41 @@ def download_cgns_structured(load=True):  # pragma: no cover
     return pyvista.get_reader(filename).read()
 
 
+def download_tecplot_ascii(load=True):  # pragma: no cover
+    """Download the single block ASCII Tecplot dataset.
+
+    Originally downloaded from Paul Bourke
+    <http://paulbourke.net/dataformats/tp/sample.tp>
```
Comment (akaszynski) [inline pyvista/examples/downloads.py:3947]:
```suggestion
    Originally downloaded from Paul Bourke's
    `Sample file <http://paulbourke.net/dataformats/tp/sample.tp>`_
```

File: pyvista/_vtk.py [hunk 4]
```diff
@@ -297,6 +297,7 @@
         vtkPTSReader,
         vtkSTLReader,
         vtkSTLWriter,
+        vtkTecplotReader,
```
Comment (MatthewFlamm) [inline pyvista/_vtk.py:300]: How do these changes pass the vtk==8.2 test if this isn't imported for vtk<9?

**Review by MatthewFlamm — COMMENTED** [review]
Also needs to be added to https://github.com/pyvista/pyvista/blob/main/doc/api/readers/index.rst#reader-classes

File: pyvista/_vtk.py (hunk 4 above)
Comment (akaszynski) [reply pyvista/_vtk.py:300]: With an install of VTK 8.1.2
```
$ pip list | grep vtk
vtk                           8.1.2
```

I can still import this.
```
>>> from pyvista import _vtk
>>> _vtk.vtkTecplotReader()
(vtkIOGeometryPython.vtkTecplotReader)0x7fbe07fb2360
```

This class appears to predate v9.1 and is imported with `from vtk import *`.

**Comment by akaszynski** [comment]
> Also needs to be added to https://github.com/pyvista/pyvista/blob/main/doc/api/readers/index.rst#reader-classes

Added in [a8e43fe](https://github.com/pyvista/pyvista/pull/2875/commits/a8e43fee74c18469e42325ef3fea57880041333f). Sorted the lines as well.

File: doc/api/readers/index.rst [hunk 5]
```diff
@@ -65,7 +66,7 @@ Reader Classes
     XMLRectilinearGridReader
     XMLStructuredGridReader
     XMLUnstructuredGridReader
-
+
```
Comment (akaszynski) [inline doc/api/readers/index.rst:69]:
```suggestion
```

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**